synapse/rest/client/v1/login.py

```diff
@@ -154,13 +154,28 @@ def _get_qualified_user_id(self, identifier):
     async def _do_appservice_login(
         self, login_submission: JsonDict, appservice: ApplicationService
     ):
-        logger.info(
-            "Got appservice login request with identifier: %r",
-            login_submission.get("identifier"),
-        )
+        identifier = login_submission.get("identifier")
+        logger.info("Got appservice login request with identifier: %r", identifier)
 
-        identifier = convert_client_dict_legacy_fields_to_identifier(login_submission)
-        qualified_user_id = self._get_qualified_user_id(identifier)
+        if not isinstance(identifier, dict):
+            raise SynapseError(
+                400, "Invalid identifier in login submission", Codes.INVALID_PARAM
+            )
+
+        # this login flow only supports identifiers of type "m.id.user".
```

Comment:
Seems like the most of this is equivalent to `_get_qualified_user_id`?

Reply:
sorry, yes it is - but I want to move the other usage of `_get_qualified_user_id` out of this class, so since it's not very big and doesn't entirely make sense as a standalone function, I just want to inline it.

Reply:
Ah, that makes more sense. 👍 I don't see an issue in inlining it, was just confused. 😄